ark/mibi/tiling_utils_test.py

```diff
@@ -465,125 +429,328 @@ def test_create_tiled_regions_non_tma(randomize_setting, moly_run, moly_interval
-        # both runs are randomized
-        else:
-            # ensure the random center points for fov 1 contain the same elements
-            # as its sorted version
-            misc_utils.verify_same_elements(
-                computed_center_points=center_points[:fov_1_end],
-                actual_center_points=actual_center_points_sorted[:fov_1_end]
+    # generate the list of automatically-generated fovs
+    auto_sample_fovs = dict(zip(auto_fov_names, auto_coords))
+
+    # define the coordinates and fov names proposed by the user
+    manual_coords = [(0, 25), (50, 25), (50, 50), (75, 50), (100, 25)]
+    manual_fov_names = ['row%d_col%d' % (x, y) for (x, y) in manual_coords]
+
+    # generate the list of manual fovs
+    manual_sample_fovs = test_utils.generate_sample_fovs_list(
+        manual_coords, manual_fov_names
+    )
+
+    # generate the mapping from manual to automatically-generated
+    manual_to_auto_map, manual_fovs_info, auto_fovs_info = \
+        tiling_utils.assign_closest_fovs(
+            manual_sample_fovs, auto_sample_fovs
+        )
+
+    # for each manual fov, ensure the centroids are the same in manual_fovs_info
+    for fov in manual_sample_fovs['fovs']:
+        # skip the Moly points
+        if fov['name'] != 'MoQC':
+            manual_centroid = tiling_utils.convert_microns_to_pixels(
+                tuple(fov['centerPointMicrons'].values())
             )
 
-            # however, fov 1 sorted entries should NOT equal fov 1 random entries
-            # NOTE: due to randomization, this test will fail once in a blue moon
-            assert center_points[:fov_1_end] != actual_center_points_sorted[:fov_1_end]
+            assert manual_fovs_info[fov['name']] == manual_centroid
 
-            # ensure the random center points for fov 2 contain the same elements
-            # as its sorted version
-            misc_utils.verify_same_elements(
-                computed_center_points=center_points[fov_1_end:],
-                actual_center_points=actual_center_points_sorted[fov_1_end:]
+    # same for automatically-generated fovs
+    for fov in auto_sample_fovs:
+        if fov != 'MoQC':
+            auto_centroid = tiling_utils.convert_microns_to_pixels(
+                auto_sample_fovs[fov]
             )
 
-            # however, fov 2 sorted entries should NOT equal fov 2 random entries
-            # NOTE: due to randomization, this test will fail once in a blue moon
-            assert center_points[fov_1_end:] != actual_center_points_sorted[fov_1_end:]
+            assert auto_fovs_info[fov] == auto_centroid
+
+    # assert the mapping is correct
+    actual_map = {
+        'row0_col25': 'row0_col0',
+        'row50_col25': 'row0_col0',
+        'row50_col50': 'row0_col100',
+        'row75_col50': 'row100_col100',
+        'row100_col25': 'row100_col0'
+    }
+
+    assert manual_to_auto_map == actual_map
```

Comment:
We should add an automatic FOV that isn't mapped to any manual FOV because it isn't the closest. 

We should also add two manual FOVs that map to the same automatic FOV. This indicates a problem, but in this case we want to make sure that it gets propagated forward, so it can be identified by the user.

Reply:
I can add a couple extra centroid coordinates in the `auto` list for the former.

For the latter, if I'm not mistaken, I think in the mapping above both `'row0_col25'` and `'row50_col25'` maps to `'row0_col0'`. Is this the test we're looking for or something else?

Reply:
Great, sounds good. Yeah that's what I had in mind.